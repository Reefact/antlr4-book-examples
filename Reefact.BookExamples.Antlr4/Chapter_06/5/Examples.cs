#region Usings declarations

using ApprovalTests;
using ApprovalTests.Reporters;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_06._5 {

    [UseReporter(typeof(VisualStudioReporter))]
    public class Examples {

        [Fact]
        [GraphicalTree("ParseTree1.svg")]
        public void mermaid_style_tree() {
            // Setup
            GRun grun = GRun.ReadResource("t.R", 6, 5);
            // Exercise
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(mermaidStyleTree);
        }

        [Fact]
        [GraphicalTree("ParseTree2.svg")]
        public void ajay_shah_b1_example() {
            // Setup
            GRun grun = GRun.ReadResource("b1.R", 6, 5);
            // Exercise
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(mermaidStyleTree);
        }

        [Fact]
        public void ajay_shah_b5_example() {
            // Setup
            GRun grun = GRun.ReadResource("b5.R", 6, 5);
            // Exercise
            string tokensString = grun.ToTokensString();
            // Verify
            Approvals.Verify(tokensString);
        }

    }

}