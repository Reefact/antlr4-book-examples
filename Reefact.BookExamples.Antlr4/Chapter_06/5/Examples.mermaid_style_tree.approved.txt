graph TD
	1["prog"] --> 2["expr_or_assign"]
	2 --> 3["expr"]
	3 --> 4["addMe"]
	2 --> 5["#0060;-"]
	2 --> 6["expr_or_assign"]
	6 --> 7["expr"]
	7 --> 8["function"]
	7 --> 9["("]
	7 --> 10["formlist"]
	10 --> 11["form"]
	11 --> 12["x"]
	10 --> 13[","]
	10 --> 14["form"]
	14 --> 15["y"]
	7 --> 16[")"]
	7 --> 17["expr"]
	17 --> 18["{"]
	17 --> 19["exprlist"]
	19 --> 20["expr_or_assign"]
	20 --> 21["expr"]
	21 --> 22["expr"]
	22 --> 23["return"]
	21 --> 24["("]
	21 --> 25["sublist"]
	25 --> 26["sub"]
	26 --> 27["expr"]
	27 --> 28["expr"]
	28 --> 29["x"]
	27 --> 30["+"]
	27 --> 31["expr"]
	31 --> 32["y"]
	21 --> 33[")"]
	17 --> 34["}"]
	1 --> 35["#0092;r#0092;n"]
	1 --> 36["expr_or_assign"]
	36 --> 37["expr"]
	37 --> 38["expr"]
	38 --> 39["addMe"]
	37 --> 40["("]
	37 --> 41["sublist"]
	41 --> 42["sub"]
	42 --> 43["x"]
	42 --> 44["="]
	42 --> 45["expr"]
	45 --> 46["1"]
	41 --> 47[","]
	41 --> 48["sub"]
	48 --> 49["expr"]
	49 --> 50["2"]
	37 --> 51[")"]
	1 --> 52["#0092;r#0092;n"]
	1 --> 53["expr_or_assign"]
	53 --> 54["expr"]
	54 --> 55["r"]
	53 --> 56["#0060;-"]
	53 --> 57["expr_or_assign"]
	57 --> 58["expr"]
	58 --> 59["expr"]
	59 --> 60["1"]
	58 --> 61[":"]
	58 --> 62["expr"]
	62 --> 63["5"]
	1 --> 64["#0060;EOF#0062;"]
